src/app/model/User.ts

```diff
@@ -27,6 +27,34 @@ const defaultUserData: UserInterface = {
 class User extends BaseModel {
   VolunteerStatus = VolunteerStatus;
 
+  /**
+   * Given a displayName returns an user id
+   * @param {string} displayName : displayName of user
+   * @return {string}
+   */
+  async getIdByDisplayName(displayName: string): Promise<string | undefined> {
+    let collection = this.getCollection("users");
+    let doc;
+    try {
+      doc = await collection.where("displayName", "==", displayName).get();
+    } catch (error) {
+      //TODO show error message to user
+      throw error;
+    }
+
+    if (doc.empty) {
+      throw Error(`This user: ${displayName} does not exist`);
+    }
+
+    doc.forEach((data) => {
```

Comment:
This is for my own understanding: why are we checking for each item in doc?

Reply:
Oh, that's a mistake. Just copied that part from the existing functions and didn't think about (or remove) it.